### PR TITLE
web: Show prompts to user when game quits

### PR DESCRIPTION
**Pull request by Andykl**
This is second issue of #4158. Probably needs a better wording.

### CODE REVIEW

File: renpy/bootstrap.py [hunk 1]
```diff
@@ -306,6 +312,14 @@ def bootstrap(renpy_base):
             except Exception as e:
                 renpy.error.report_exception(e)
 
+        if renpy.emscripten:
+            import emscripten
+            if exit_status == 0:
+                emscripten.run_script('Module.print("The game quits. You may close the tab.")')
```
Comment (Gouvernathor) [inline renpy/bootstrap.py:318]: Maybe these should allow translations, using something like this:
```diff
-                 emscripten.run_script('Module.print("The game quits. You may close the tab.")')
+                 emscripten.run_script('Module.print("{}")'.format(__("The game quits. You may close the tab.")))
```
That way the messages, and them only, could be translated through common.rpym without risk of typos in the emscripten call.

File: renpy/bootstrap.py [hunk 2]
```diff
@@ -292,7 +292,13 @@ def bootstrap(renpy_base):
                 exit_status = e.status
 
                 if e.relaunch:
-                    if hasattr(sys, "renpy_executable"):
+                    if renpy.emscripten:
+                        import emscripten
+                        emscripten.run_script('Module.print("The game asks for a restart. '
```
Comment (qinlili23333) [inline renpy/bootstrap.py:297]:
```
emscripten.run_script('location.reload()')
```
We can just automatically reload the webpage.

**Comment by renpytom** [comment]
I've implemented something similar in renpy/renpy-build@0b1daac , except it's done at the Python level, so it'll catch most ways of exiting. 

In general, a web game should never quit. Web games are similar to mobile games, insofar as quitting is an error.

**Comment by qinlili23333** [comment]
> In general, a web game should never quit. Web games are similar to mobile games, insofar as quitting is an error.

However lots of mobile games on Android have implement quitting actions just like the picture below.
Screenshot_20221219-105425_War Robots

In my view, platforms with a "Back" action feature do need quitting feature. When user click "Back" at homepage, what user expect is the game can quit so they don't need to manually kill it from recent activities. 
Web apps on Android and ChromeOS accept the "Back" action and should work as native app so web game should have quit feature to provide a better PWA user experience.

We can implement quitting under specify conditions:
```
if (history.length == 1 && window.self === window.top) {
    window.close();
};
```
`window.close()` only works when no history to go back and it's not in iframe, so we only need to implement quitting in this condition.

**Pull request closed without merging**